.gitlab-ci.yml

```diff
@@ -23,17 +20,19 @@ cache:
 
 before_script:
   - python -V  # Print out python version for debugging
-  - pip install -r requirements.txt
-  - pip install python-coveralls
+  - pip install -r requirements.txt python-coveralls pylint pycodestyle
 
 test:
   stage: test
   script:
-    - echo $TEST_SUITE
-    - echo $TEST_SUITE > test.sh
-    - bash test.sh
-    - coverage report
-  #  - coveralls
+    # TODO bjuhasz: use tox because "setup.py test" is deprecated:
+    # https://setuptools.readthedocs.io/en/latest/setuptools.html#test-build-package-and-run-a-unittest-suite
+    - coverage run --branch --source=pyega3 setup.py test
+    - coverage report --fail-under=$COVERAGE_FAIL_UNDER
+    - pylint --ignore=test/support --fail-under=$PYLINT_FAIL_UNDER pyega3/ test/
+    - pycodestyle --show-source --show-pep8 --statistics --count --max-line-length=120 --exclude=test/support,.git,__pycache__
```

Comment:
`test/support`has been removed, so that argument can be simplified to `--exclude=.git,__pycache__`